### ISSUE TITLE
MAUI Testability

### ISSUE THREAD

**Issue by Kebechet** [Kebechet/Maui.RevenueCat.InAppBilling#8]
Right now when you add `InAppBilling` Nuget into your MAUI project, the project wont be testable. 
Because of: 

> error NU1202: Package Kebechet.Maui.RevenueCat.InAppBilling 1.0.5 is not compatible with net7.0 (.NETCoreApp,Version=v7.0). Package Kebechet.Maui.RevenueCat.InAppBilling 1.0.5 supports:
> error NU1202:   - net7.0-android33.0 (.NETCoreApp,Version=v7.0)
> error NU1202:   - net7.0-ios16.1 (.NETCoreApp,Version=v7.0)
> error NU1202:   - net7.0-maccatalyst16.1 (.NETCoreApp,Version=v7.0)
> error NU1202:   - net7.0-windows10.0.19041 (.NETCoreApp,Version=v7.0)

When I tried to add `.NET7` as another `TargetFramework` I got:
`<X> must have an implementation part because it has accessibility modifiers.`

The reason behind it is that main MAUI app has to add pure `.NET7` framework, so that unit-test library can link it. But this is not currently possible in library projects like this one. Apparently this should be fixed in .NET8:
https://github.com/dotnet/maui/issues/3552
https://github.com/dotnet/maui/discussions/14847

**Comment by Kebechet** [comment]
https://www.linkedin.com/posts/jakub-florkowski_what-happens-when-you-let-ai-agents-build-share-7455575114110173184-lIlK?utm_source=share&utm_medium=member_desktop&rcm=ACoAAC3wUhcBS50BuAyaMP3t7K06dAORrRT6CnQ

MauiDevFlow

**Comment by Kebechet** [comment]
Note: the original TFM blocker is largely resolved by the move to .NET 9 — the wrapper csproj now includes a vanilla `net9.0` target alongside the platform-specific ones (`src/Maui.RevenueCat.InAppBilling/Maui.RevenueCat.csproj:4`), so a plain `net9.0` xUnit project can `ProjectReference` it today. Repurposing this issue to track what kind of tests to actually add.

Two complementary approaches worth pursuing:

## 1. Public API snapshot test (cheap, high value, CI-friendly)

[`PublicApiGenerator`](https://github.com/PublicApiGenerator/PublicApiGenerator) + [`Verify`](https://github.com/VerifyTests/Verify) generates a textual snapshot of the wrapper's entire public surface and fails CI whenever the `.received.txt` diverges from the committed `.verified.txt`.

- Single `net9.0` xUnit project, ~30 lines of code, no MAUI workloads, no device, no store account.
- Every breaking change (e.g. removing `PricingDto.OriginalPrice*` per #108, retyping `OfferingDto.Metadata` per #110) lands as a reviewable diff against the committed snapshot — impossible to merge silently.
- Pattern used by Polly, Serilog, MediatR, FluentValidation.
- Should land *before* #108 / #110 so those breaking changes ship with their first reviewable API diff.

Also worth a compile-only consumer-smoke project, similar to the existing `tests/BindingLinkerCheck/` for the iOS binding — a tiny `net9.0-android` + `net9.0-ios` exe that ProjectReferences the wrapper and touches every public DTO / `IRevenueCatBilling` method. Catches signature changes that pure `net9.0` API surface checks miss (e.g. extension methods that only exist under `#if IOS`).

## 2. End-to-end DemoApp automation via [MauiDevFlow](https://github.com/Redth/MauiDevFlow) (longer-term, runtime regression coverage)

Redth's MauiDevFlow is an agentic dev-loop tool (build → deploy → screenshot → tap → inspect → fix). Not a unit-testing framework, and explicitly `#if DEBUG`-only / not shipping in apps — but it's a fit for end-to-end smoke-testing the `demo/DemoApp/` project after wrapper changes:

- After an SDK bump (Android RevenueCat / iOS binding regen) or a wrapper-level change, an agent builds the demo for an iOS sim + Android emulator, drives the "Load Offerings" / "Purchase" / "Restore" flows, screenshots the result, asserts nothing crashed.
- Catches runtime regressions API-surface checks miss — e.g. an Android extension throwing because we changed a mapping, or an iOS binding linker error only surfacing in a consumer build.
- Needs the demo to be CI-runnable: sandbox tester account, RC API key as a secret, network. Setup cost is real.
- Best ROI is around the iOS binding regen pipeline and SDK version bumps — i.e. the changes most likely to break at *runtime* rather than at compile time.

## Suggested ordering

1. Land approach 1 (API snapshot + binding-style consumer smoke) — small, immediate, protects #108 / #110.
2. Defer approach 2 to a follow-up issue once 1 is in place — bigger setup, niche-but-real value for the binding regen path.
